File: lesson4/c_c++/homework/homework4_31.cpp
//: lesson4/c_c++/homework/homework4_31.cpp
#include <iostream>

int main() {
    int numb_n = 100000;
    int numb_i;
    for (numb_i = 2; numb_i < numb_n; numb_i++) {
        int numb_s = 0;
        for (int numb_j = 1; numb_j < numb_i; numb_j++) {
            if (numb_i % numb_j == 0) {
                numb_s += numb_j;
            }
        }
        if (numb_s == numb_i) {
            std::cout << "perfect => " << numb_i << '\n';
        }
    }
    return 0;
}

File: lesson4/python3/homework/homework4_31.py
#!/usr/bin/python3
#: lesson4/python3/homework/homework4_31.py
# Напишите программу, которая вводит натуральное число N и находит все
# совершенные числа в диапазоне от 1 до N.

for numb_i in range(2, 50000):
    numb_s = 0
    for numb_j in range(1, numb_i):
        if numb_i % numb_j == 0:
            numb_s += numb_j
    if numb_s == numb_i:
        print("совершенное число =>", numb_i)

